### ISSUE TITLE
ui5-input suggestions don't show if you change the value programmatically

### ISSUE THREAD

**Issue by vladitasev** [SAP/ui5-webcomponents#4029]
**Reproduction**:

```html
<ui5-input id="inp" show-suggestions>
		<ui5-suggestion-item text="Adam D" description="Administrative Support"></ui5-suggestion-item>
		<ui5-suggestion-item text="Aanya Sing" description="Administrative Support"></ui5-suggestion-item>
		<ui5-suggestion-item text="Allen K" description="Technical Support"></ui5-suggestion-item>
		<ui5-suggestion-item text="Alex" description="Technical Support"></ui5-suggestion-item>
	</ui5-input>

	<ui5-button id="btn">Clear</ui5-button>

	<script>
		btn.addEventListener("click", () => {
			inp.value = "";
		});
	</script>
```

**Actual behavior:**

1. Click the input, the suggestions popup will appear, click a suggestion, it will be set as the input value
2. Delete the input value with the keyboard (f.e. backspace)
3. Click the input again, suggestions work (the popup shows). If you click outside and then back in the input, suggestions work.
4. Now click the "Clear" button to do the same as in step 2, but programmatically.
5. Click the input, the popup does not show any more. Click outside, back in, no matter what you do the popup is gone. The only way to bring the popup back is to type something and delete it.

**Expected behavior:** The input with suggestions should behave the same way no matter how you modify the value.

This kind of bug generally means that there is something done imperatively on user input. Instead, the rendering should be entirely state-based. When the user does something, only the state should be changed and the component will re-render itself properly.

**Comment by Todor-ads** [comment]
Hi @SAP/ui5-webcomponents-topic-rl
The Input component is part of your topic.
Isolated example - https://codesandbox.io/s/compassionate-ganguly-9ox9c?file=/src/index.js

Best Regards,
Todor

**Comment by niyap** [comment]
The issue should be fixed with the following pull request: #3976
